Sources/MongoSwift/Operations/ChangeStreamNextOperation.swift

```diff
@@ -0,0 +1,44 @@
+import mongoc
+
+/// An operation corresponding to a `next` call on a ChangeStream.
+internal struct ChangeStreamNextOperation<T: Codable>: Operation {
+    private let changeStream: ChangeStream<T>
+
+    internal init(changeStream: ChangeStream<T>) {
+        self.changeStream = changeStream
+    }
+
+    internal func execute(using connection: Connection, session: ClientSession?) throws -> T? {
+        // We already check this in `ChangeStream.next()` in order to extract the relevant connection and session,
+        // but error again here just in case.
+        guard case let .open(changeStreamPtr, _, _, _) = changeStream.state else {
+            throw ClosedChangeStreamError
+        }
+
```

Comment:
comparing to the `NextOperation` for cursors, I noticed we are doing a check there to see if the session is still active and throwing an error if not. I think we probably need to check that here as well, what do you think @patrickfreed ?